src/data/repositories/IncidentManagementTeamD2Repository.ts

```diff
@@ -0,0 +1,254 @@
+                    )
+                    .map(d2User =>
+                        this.mapUserToIncidentManagementTeamMember(
+                            d2User as D2UserFix,
+                            events,
+                            dataElementRoles
+                        )
+                    );
+            }
+        );
+    }
+
+    private mapUserToIncidentManagementTeamMember(
+        d2User: D2UserFix,
+        events: D2TrackerEvent[],
+        dataElementRoles: D2DataElement[]
+    ): TeamMember {
+        const avatarId = d2User?.avatar?.id;
+        const photoUrlString = avatarId
+            ? `${this.api.baseUrl}/api/fileResources/${avatarId}/data`
+            : undefined;
+
+        const teamMember = new TeamMember({
+            id: d2User.id,
+            username: d2User.username,
+            name: d2User.name,
+            email: d2User.email,
+            phone: d2User.phoneNumber,
+            status: "Available", // TODO: Get status when defined
+            photo:
+                photoUrlString && TeamMember.isValidPhotoUrl(photoUrlString)
+                    ? new URL(photoUrlString)
+                    : undefined,
+            teamRoles: undefined,
+            workPosition: undefined, // TODO: Get workPosition when defined
+        });
+
+        const teamRoles = getTeamMemberIncidentManagementTeamRoles(
+            teamMember,
+            events,
+            dataElementRoles
+        );
+
+        return new TeamMember({
+            ...teamMember,
+            teamRoles: teamRoles.length > 0 ? teamRoles : undefined,
+        });
+    }
+
+    saveIncidentManagementTeamMemberRole(
+        teamMemberRole: TeamRole,
+        incidentManagementTeamMember: TeamMember,
+        diseaseOutbreakId: Id
+    ): FutureData<void> {
+        return this.saveOrDeleteIncidentManagementTeamMember(
+            teamMemberRole,
+            incidentManagementTeamMember,
+            diseaseOutbreakId,
+            "CREATE_AND_UPDATE"
+        );
+    }
+
+    deleteIncidentManagementTeamMemberRole(
+        teamMemberRole: TeamRole,
+        incidentManagementTeamMember: TeamMember,
+        diseaseOutbreakId: Id
+    ): FutureData<void> {
+        return this.saveOrDeleteIncidentManagementTeamMember(
+            teamMemberRole,
+            incidentManagementTeamMember,
+            diseaseOutbreakId,
+            "DELETE"
+        );
+    }
+
+    private getDataElementRolesAndIncidentManagementTeamEvents(diseaseOutbreakId: Id): FutureData<{
+        dataElementRoles: D2DataElement[];
+        events: D2TrackerEvent[];
+    }> {
+        return Future.joinObj(
+            {
+                dataElementRoles: apiToFuture(
+                    this.api.models.dataElements.get({
+                        fields: dataElementFields,
+                        paging: false,
+                        filter: {
+                            id: {
```

Comment:
If you want to get the data elements of incident management team builder program stage you can use : https://metadata.eyeseetea.com/rsl-zebra/api/programStages/DwEOpUBGgOp/programStageDataElements
If you do it from the above api, then if new data elements are added to the program stage they will automatically be fetched. I am not sure if that will help in this context, if it does, then you can use it.